doc/life-cycles/digestxof.dot

```diff
@@ -0,0 +1,39 @@
+digraph digestxof {
+    bgcolor="transparent";
+
+    begin [label=start, color="#deeaee", style="filled"];
+    newed [label=newed, fontcolor="#c94c4c", style="solid"];
+    initialised [label=initialised, fontcolor="#c94c4c"];
+    absorbed [label="absorbed", fontcolor="#c94c4c"];
+    finaled [label="finaled", fontcolor="#c94c4c"];
```

Comment:
I'm unsure that this state needs to exist.
Could EVP_DigestFinalXOF be the same as the squeeze function?

I.e. what's the rationale for having both?

Reply:
It depends on if Final is a one shot operation that cant be followed by a squeeze.